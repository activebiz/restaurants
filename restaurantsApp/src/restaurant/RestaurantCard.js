import React from 'react';
import {StyleSheet, View, Linking, Alert} from 'react-native';
import {Card} from 'react-native-elements';
import {Chip} from 'react-native-elements';
import Icon from 'react-native-vector-icons/MaterialIcons';
import Address from './Address';

const styles = StyleSheet.create({
  addressContainer: {
    flexDirection: 'row',
  },
  addressItem: {
    marginLeft: 'auto',
  },
  addressAction: {
    marginLeft: 'auto',
  },
});

const RestaurantCard = props => {
  const {name, url, geo: {address = {}} = {}} = props;

  const openUrl = async value => {
    try {
      const supported = await Linking.canOpenURL(value);
      if (!supported) {
        return Promise.reject(new Error('opening url is not supported'));
      }
      return await Linking.openURL(value);
    } catch (err) {
      console.log(`Error opening url ${err}`);
      Alert.alert('could not open url');
    }
  };

  return (
    <Card>
      <Card.Title>
        <Chip
          title={name}
          icon={{
            name: 'restaurant',
            type: 'MaterialIcons',
            size: 24,
            color: 'white',
          }}
          iconRight
        />
      </Card.Title>
      <Card.Divider />
      <View style={styles.addressContainer}>
        <Address style={styles.addressItem} {...address} />
        <Icon
          style={styles.addressAction}
          reverse
          name="launch"
          size={24}
          onPress={() => openUrl(url)}
        />
      </View>
    </Card>
  );
};

export default RestaurantCard;
