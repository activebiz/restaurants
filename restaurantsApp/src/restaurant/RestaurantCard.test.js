import 'react-native';
import React from 'react';
import RestaurantCard from './RestaurantCard';
import renderer from 'react-test-renderer';

describe('RestaurantCard', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<RestaurantCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
