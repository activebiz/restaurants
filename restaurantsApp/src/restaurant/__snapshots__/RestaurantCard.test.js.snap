// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantCard renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 0,
      "padding": 15,
      "shadowColor": "rgba(0,0,0, .2)",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#43484d",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 15,
          "textAlign": "center",
        }
      }
      testID="cardTitle"
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 3,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 3,
            },
            undefined,
            false,
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 30,
              "borderWidth": 0,
              "flexDirection": "row-reverse",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <View
            style={
              Object {
                "marginHorizontal": 5,
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 24,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="iconIcon"
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#bcbbc1",
          "height": 0.5,
          "marginBottom": 15,
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View>
        <Text />
        <Text />
        <Text />
      </View>
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        reverse={true}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 24,
            },
            Object {
              "marginLeft": "auto",
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
